regulation-worker/internal/client/client.go

```diff
@@ -12,105 +12,133 @@ import (
 	"time"
 
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
 )
 
+var pkgLogger = logger.NewLogger().Child("client")
+
 type JobAPI struct {
+	Client         *http.Client
+	WorkspaceID    string
 	URLPrefix      string
 	WorkspaceToken string
-	WorkspaceID    string
 }
 
 //Get sends http request with workspaceID in the url and receives a json payload
 //which is decoded using schema and then mapped from schema to internal model.Job struct,
 //which is actually returned.
 func (j *JobAPI) Get(ctx context.Context) (model.Job, error) {
 
+	pkgLogger.Debugf("making http request to regulation manager to get new job")
 	ctx, cancel := context.WithTimeout(ctx, time.Duration(time.Minute))
 	defer cancel()
 
 	method := "GET"
 	genEndPoint := "/dataplane/workspaces/{workspace_id}/regulations/workerJobs"
 	url := fmt.Sprint(j.URLPrefix, prepURL(genEndPoint, j.WorkspaceID))
+	pkgLogger.Debugf("making GET request to URL: %v", url)
 
 	req, err := http.NewRequestWithContext(ctx, method, url, nil)
 	if err != nil {
+		pkgLogger.Errorf("error while create new http request: %v", err)
 		return model.Job{}, err
 	}
 
 	req.SetBasicAuth(j.WorkspaceToken, "")
 	req.Header.Set("Content-Type", "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	pkgLogger.Debugf("making request: %v", req)
```

Comment:
We already do 	`pkgLogger.Debugf("making GET request to URL: %v", url)` one of those are redundant.